server/vulnerabilities/vuln_centos/centos.go

```diff
@@ -0,0 +1,512 @@
+			for cve := range cves {
+				allPkgs.Add(pkg, cve)
+			}
+		}
+	}
+
+	return allPkgs, nil
+}
+
+func processRepoMD(mdURL url.URL, localDir string, verbose bool) error {
+	resp, err := http.Get(mdURL.String())
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+	type location struct {
+		Href string `xml:"href,attr"`
+	}
+	type repoDataItem struct {
+		Type     string   `xml:"type,attr"`
+		Location location `xml:"location"`
+	}
+	type repoMetadata struct {
+		XMLName xml.Name       `xml:"repomd"`
+		Datas   []repoDataItem `xml:"data"`
+	}
+	var md repoMetadata
+	if err := xml.Unmarshal(b, &md); err != nil {
+		return err
+	}
+	for _, data := range md.Datas {
+		if data.Type != "primary_db" && data.Type != "other_db" {
+			continue
+		}
+		sqliteURL := mdURL
+		sqliteURL.Path = strings.TrimSuffix(sqliteURL.Path, "repomd.xml") + strings.TrimPrefix(data.Location.Href, "repodata/")
+		if err := downloadSqlite(sqliteURL, localDir, verbose); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func downloadSqlite(sqliteURL url.URL, localDir string, verbose bool) error {
+	if verbose {
+		fmt.Printf("%s\n", sqliteURL.Path)
+	}
+	filePath := filepath.Join(localDir, sqliteURL.Path)
+	filePath = strings.TrimSuffix(filePath, ".bz2")
+	filePath = strings.TrimSuffix(filePath, ".xz")
+	_, err := os.Stat(filePath)
+	switch {
+	case err == nil:
+		return nil
+	case errors.Is(err, os.ErrNotExist):
+		// OK
+	default:
+		return err
+	}
+
+	resp, err := http.Get(sqliteURL.String())
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if err := os.MkdirAll(filepath.Dir(filePath), 0o755); err != nil {
+		return err
+	}
+	tmpFile, err := ioutil.TempFile("", fmt.Sprintf("%s*", filepath.Base(filePath)))
+	if err != nil {
+		return err
+	}
+	defer tmpFile.Close()
+
+	var decompressor io.Reader
+	switch {
+	case strings.HasSuffix(sqliteURL.Path, "bz2"):
+		decompressor = bzip2.NewReader(resp.Body)
+	case strings.HasSuffix(sqliteURL.Path, "xz"):
+		decompressor, err = xz.NewReader(resp.Body)
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("unknown extension: %s", sqliteURL.Path)
+	}
+	if _, err := io.Copy(tmpFile, decompressor); err != nil {
+		return err
+	}
+	if err := tmpFile.Close(); err != nil {
+		return err
+	}
+	if err := os.Rename(tmpFile.Name(), filePath); err != nil {
+		return err
+	}
+	return nil
+}
+
+func processSqlites(dbPaths dbs) (CentOSPkgSet, error) {
+	db, err := sql.Open("sqlite3", dbPaths.primary)
+	if err != nil {
+		return nil, err
+	}
+	defer db.Close()
+
+	if _, err := db.Exec(fmt.Sprintf("ATTACH DATABASE '%s' as other;", dbPaths.other)); err != nil {
+		return nil, err
+	}
+
+	db.SetMaxOpenConns(1)
+	rows, err := db.Query(`SELECT
+		p.name, p.version, p.release, p.arch, c.changelog
+		FROM packages p
+		JOIN other.changelog c ON (p.pkgKey=c.pkgKey)
+		WHERE c.changelog LIKE '%CVE-%-%';`)
```

Comment:
This is simply just matching any changelog entry with a mention of a "CVE-...-...".

I sampled some findings and they seem to be good catches. So far I haven't seen any "This release does not patch CVE-...-..." in the changelogs :)

Reply:
Why not `'%CVE-%'` ? It might not be a huge difference, but I feel it should match similarly and a bit faster.